src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -5070,11 +5070,7 @@ internal static void InitializeSockets()
                 {
                     if (!s_initialized)
                     {
-                        // TODO (#7849): this call is not required for *NIX and should be avoided during PAL design.
-
-                        // Ensure that WSAStartup has been called once per process.  
-                        // The System.Net.NameResolution contract is responsible with the initialization.
-                        Dns.GetHostName();
+                        SocketPal.Initialize();
```

Comment:
Just curious: Since the Unix implementation of `Initialize` is a no-op, could `Initialize` be a partial method on `Socket`? `Socket` is already a partial class and there are already separate `Socket.Windows.cs` and `Socket.Unix.cs` files. Then `Initialize` wouldn't be called at all on Unix and there wouldn't be a need for a no-op implementation of `Initialize` in `SocketPal.Unix.cs`. I guess it doesn't matter that much either way since `Initialize` is only called once.